[core] Batch small changes (#21928)
mui-org/material-ui: docs/src/pages/discover-more/roadmap/roadmap.md

- 0.2 - **TypeScript.** 📏 There are two dimensions to this problem:
  - The definitions. We are **continuously** improving them. For instance, we are working on moving the props descriptions to TypeScript, so you can access them directly from your IDE.
    The codebase is written in JavaScript, we don't plan on migrating it to TypeScript in the near future. Upvote [#15984](https://github.com/mui-org/material-ui/issues/15984) if you want us to rewrite the core in a future version.
- 0.2 - **Accessibility.** ♿️ We have relatively [few accessibility issues](https://darekkay.com/blog/accessible-ui-frameworks/), but we are eager to address them all. We would appreciate the help of accessibility experts.
- 0.2 - **Material Design Update.** 🎀 We are relatively up-to-date but the material design specification [is evolving](https://material.io/whats-new/), so should we.

## New components

Here are the components we will work on being supported in the Material-UI ecosystem, no matter if they are part of the [community edition](https://github.com/mui-org/material-ui) (MIT license) or [Material-UI X](https://github.com/mui-org/material-ui-x), a paid extension for enterprises:

> - 🧪 Close to become stable, already released as unstable

Reviewer: ```suggestion
- 🧪 Close to becoming stable, already released as unstable
```

> - 🛠 Work in progress, will or already released as unstable

Reviewer: ```suggestion
- 🛠 Work in progress, will be or already released as unstable
```

- ⏳ Planning to build

> - ⭐️ We will make some of the features available in [Material-UI X](https://github.com/mui-org/material-ui-x) (non-MIT)

Reviewer 2: This symbols says "new" to me.  

```suggestion
- 💵 We will make some of the features available in [Material-UI X](https://github.com/mui-org/material-ui-x) (non-MIT)
```
Or:
🔐 💳 💎 📦 ?

Reviewer 2: ✨ - sparkly and new?
⚡ - possibly, although does it suggest "paid" to you?
💎 - says "premium", and is associated with value. Could be an option, although not to be confused with Diamond Sponsors.

Author: I would vote for ⚡, but not strong preference. I think that we could ask the designers to come with something to differentiate the different pages in the left side navigation as well on the overall layout.

| Data Grid                  | 🛠⭐️   |
| Data Grid - Column pinning | ⏳⭐️  |
| Data Grid - Nested row     | ⏳⭐️  |
| Tree View                  | 🛠      |
| Tree View - Multiselect    | 🛠⭐️   |
| Tree View - Checkbox       | ⏳⭐️  |
| Tree View - Drag & Drop    | ⏳⭐️  |
| Tree View - Virtualization | ⏳⭐️  |
| Carousel                   | ⏳     |
| Dropdown                   | ⏳     |
| Dropzone                   | ⏳     |
| File Upload                | ⏳     |
| Image                      | ⏳     |
| Stack                      | ⏳     |
| Navbar                     | ⏳     |
| NProgress                  | ⏳     |
| Nested Menu                | ⏳     |
| Numeric Input              | ⏳     |
| Scrollspy                  | ⏳     |
| TopLayout                  | ⏳     |
| Splitter                   | ⏳⭐️  |
| Masonry                    | ⏳⭐️  |
| Pivot Grid                 | ⏳⭐️  |
| Big Calendar               | ⏳⭐️  |
| Sparkline                  | ⏳⭐️  |
| Gauge                      | ⏳⭐️  |
| Charts                     | ⏳⭐️  |
| TreeSelect                 | ⏳⭐️  |
| Rich Text Editor           | ⏳⭐️  |
| Gantt Chart                | ⏳⭐️  |
| Color Picker               | ⏳⭐️  |

> > ⚠️ **Disclaimer**: We operate in a dynamic environment, and things are subject to change. The information provided is intended to outline the general framework direction. It's intended for informational purposes only. We may decide to add/remove new items at any time depending on our capability to deliver while meeting our quality standards. The development, releases and timing of any features or functionality of Material-UI remains at the sole discretion of Material-UI. The roadmap does not represent a commitment, obligation or promise to deliver at any time.

Reviewer 2: ```suggestion
> ⚠️ **Forward looking statement.** We operate in a dynamic environment, and things are subject to change. The information provided here is intended to outline the general framework direction. It's intended for informational purposes only. We may decide to add/remove new items at any time depending on our capability to deliver while meeting our quality standards. The development, releases and timing of any features or functionality of Material-UI remains at the sole discretion of Material-UI. The roadmap does not represent a commitment, obligation or promise to deliver at any time. You should not depend on the roadmap when making technology decisions.
```
